polkitbackend: Fix compilation on macOS

Some functions from netdb.h are only defined on macOS when either
_POSIX_C_SOURCE is not defined, or _DARWIN_C_SOURCE is defined.
Compiling with _XOPEN_SOURCE automatically defines _POSIX_C_SOURCE on
macOS, which hides these files.

Add -D_DARWIN_C_SOURCE=1 to the compiler flags to make these functions
available and fix the compilation issue on macOS.

Upstream-Staus: Pending
--- src/polkitbackend/meson.build	2022-01-27 17:49:49.000000000 +0100
+++ src/polkitbackend/meson.build.orig	2022-01-27 17:49:14.000000000 +0100
@@ -31,6 +31,7 @@
   '-DPACKAGE_SYSCONF_DIR="@0@"'.format(pk_prefix / pk_sysconfdir),
   '-D_XOPEN_SOURCE=700',
   '-D_GNU_SOURCE=1',
+  '-D_DARWIN_C_SOURCE=1',
 ]
 
 if js_engine == 'duktape'
